### ISSUE TITLE
Performance issue due to XRegExp.prototype modification

### ISSUE THREAD

**Issue by brandonkelly** [slevithan/xregexp#134]
Firefox is giving me this notice in the console when using xregexp:

> mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (xregexp-all.js line 2732)

Referenced line:

```js
regex.__proto__ = XRegExp.prototype;
```

**Comment by abozhilov** [comment]
True, but setting `__proto__` is quite different than `Object.create`. Built-in instances like arrays, functions, regex have internal methods. Those methods are not exposed through the prototype chain. In other words you can not extend built ins prior ES6. http://perfectionkills.com/how-ecmascript-5-still-does-not-allow-to-subclass-an-array/

In the other hand `Object.create` always returns native object. If you try to do: `var regex = Object.create(/^regex$/)`, the returned object is not exactly regex since it doesn't have internal methods and properties of a real regex instance.

**Comment by slevithan** [comment]
@abozhilov describes why this is used -- you can't subclass builtins like `RegExp` prior to ES6 / ES 2015.

In fact XRegExp already has an alternative/fallback for browsers that don't support `__proto__`. See [here](https://github.com/slevithan/xregexp/blob/37413619ea9f03638bfa92e038e4b4c15645d446/src/xregexp.js#L117-L127). But that is *actually* slower, rather than the hypothetical slowdown described in the console message.

And in fact you could remove all related code and everything would still work. XRegExp doesn't actually add anything to `XRegExp.prototype`. The point of this code is just so that developers *using* XRegExp can choose to do so if they wish.

It would be nice to start using builtin subclassing for this in browsers that support it. Then the use of `__proto__` could be removed entirely, and the existing fallback of copying over properties from `XRegExp.prototype` could cover non ES 2015 browsers.

**Comment by slevithan** [comment]
XRegExp is now mostly in maintenance mode. Its spiritual successor is [Regex+](https://github.com/slevithan/regex), which is designed for modern JavaScript, is much more lightweight, and includes additional powerful features not present in XRegExp.
